src/daemonStarter.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { join } from 'path';
+import { spawn } from 'child_process';
+import { ExitCode } from './local-ssh/common';
+import { ILogService } from './services/logService';
+
+const sleep = (sec: number) => new Promise(resolve => setTimeout(resolve, sec * 1000));
+
+export async function ensureDaemonStarted(logService: ILogService, retry = 10) {
+    if (retry < 0) {
+        return;
+    }
+    const localAppProcess = await tryStartDaemon(logService);
+    localAppProcess.once('exit', code => {
+        switch (code) {
+            case ExitCode.OK:
+            case ExitCode.ListenPortFailed:
+                logService.error('exit with code: ' + code);
+                return;
+        }
+        logService.error('unexpectedly exit with code: ' + code + ' attempt retry: ' + retry);
+        ensureDaemonStarted(logService, retry - 1);
+    });
+    await sleep(1);
+}
+
+export async function tryStartDaemon(logService: ILogService) {
+    logService.info('going to start local-ssh daemon');
+    const daemon = spawn(process.execPath, [join(__dirname, 'local-ssh/daemon.js')], {
+        detached: true,
+        stdio: 'ignore',
```

Comment:
where logs goes? It woudl be good if Export Logs command can capture it and a user coudl understand from `Gitpod` output view what is happening.

Reply:
We can't get vscode context inside daemon, even we pipe outputs into vscode outputs in `extension.ts`, the window that start this daemon may be closed.

How about add this `.log` file into `Export Logs` files?